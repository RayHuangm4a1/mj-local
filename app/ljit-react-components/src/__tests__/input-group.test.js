import React from 'react';
import { shallow, mount, } from 'enzyme';
import InputGroup from '../components/input-group';

jest.mock('antd/lib/input/Group', () => function mockComponent() {
	return <div />;
});

const { InputSizeEnums, } = InputGroup;

describe('InputGroup', () => {
	it('should handle default props', () => {
		const {
			isCompact,
			size,
			className,
			style,
		} = InputGroup.defaultProps;

		expect(isCompact).toEqual(false);
		expect(size).toEqual(InputSizeEnums.DEFAULT);
		expect(className).toEqual('');
		expect(style).toEqual({});
	});

	it('should render correctly', () => {
		const props = {
			isCompact: true,
			size: InputSizeEnums.DEFAULT,
			className: 'snapshot-class',
			style: {},
		};
		const wrapper = shallow(<InputGroup {...props}/>);

		expect(wrapper).toMatchSnapshot();
	});

	it('should be selectable by class ljit-input-group', () => {
		const wrapper = shallow(<InputGroup />);

		expect(wrapper.hasClass('ljit-input-group')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<InputGroup className={className} />);

		expect(wrapper.hasClass(className)).toEqual(true);
	});

	it('should be selectable by size: DEFAULT', () => {
		const size = InputSizeEnums.DEFAULT;
		const wrapper = shallow(<InputGroup size={size} />);

		expect(wrapper.hasClass('ljit-input-group-size--default')).toEqual(true);
	});

	it('should be selectable by class ljit-input-group-compact when isCompact equals true', () => {
		const isCompact = true;
		const wrapper = shallow(<InputGroup isCompact={isCompact} />);

		expect(wrapper.hasClass('ljit-input-group-compact')).toEqual(true);
	});

	it('should mount in a full DOM', () => {
		const inputNode = (
			<React.Fragment>
				<input placeholder="mount-placeholder-1"/>
				<input placeholder="mount-placeholder-2"/>
			</React.Fragment>
		);
		const props = {
			isCompact: true,
			size: InputSizeEnums.LARGE,
			className: 'mount-class',
			style: { width: '264px', },
			children: inputNode,
		};
		const wrapper = mount(<InputGroup {...props} />);

		expect(wrapper.props().isCompact).toBe(props.isCompact);
		expect(wrapper.props().size).toBe(props.size);
		expect(wrapper.props().className).toBe(props.className);
		expect(wrapper.props().style).toBe(props.style);
		expect(wrapper.props().children).toBe(props.children);
	});

	describe('InputSize Enums', () => {
		it('should contain DEFAULT property', () => {
			const typeName = 'default';
			const formatType = 'DEFAULT';

			expect(InputSizeEnums[formatType]).toEqual(typeName);
		});

		it('should contain LARGE property', () => {
			const typeName = 'large';
			const formatType = 'LARGE';

			expect(InputSizeEnums[formatType]).toEqual(typeName);
		});

		it('should contain SMALL property', () => {
			const typeName = 'small';
			const formatType = 'SMALL';

			expect(InputSizeEnums[formatType]).toEqual(typeName);
		});
	});
});