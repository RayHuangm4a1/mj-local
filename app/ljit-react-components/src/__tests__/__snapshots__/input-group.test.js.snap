// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputGroup
    className="snapshot-class"
    isCompact={true}
    size="default"
    style={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
      "className": "ljit-input-group ljit-input-group-size--default ljit-input-group-compact snapshot-class",
      "compact": true,
      "size": "default",
      "style": Object {},
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
        "className": "ljit-input-group ljit-input-group-size--default ljit-input-group-compact snapshot-class",
        "compact": true,
        "size": "default",
        "style": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
